paragraph/translation/allgather/bidir_ring_allgather_translator.cc

```diff
@@ -0,0 +1,125 @@
+/* Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "paragraph/translation/allgather/bidir_ring_allgather_translator.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "factory/ObjectFactory.h"
+#include "paragraph/translation/barrier/barrier_translator.h"
+
+namespace paragraph {
+
+BiDirRingAllGatherTranslator::BiDirRingAllGatherTranslator(
+    nlohmann::json config) {
+  barrier_translator_ = nullptr;
+  if (config.find("barrier") != config.end()) {
+    auto maybe_translator = BarrierTranslator::Create(config["barrier"]);
+    CHECK_OK(maybe_translator.status());
+    barrier_translator_ = std::move(maybe_translator.value());
+  }
+  // Create json config for internal unidir_ring all-gather
+  nlohmann::json implicit_config = R"(
```

Comment:
You only need:
{ "algorithm": "unidir-ring" }

No need to create more then ignore it.